src/accountsdb/snapshots.zig

```diff
@@ -2215,44 +2214,72 @@ pub const SnapshotFiles = struct {
+        defer incremental_snaps.deinit(allocator);
+
+        var maybe_latest_full: ?FullSnapshotFileInfo = null;
+
+        var dir_iter = search_dir.iterate();
+        while (try dir_iter.next()) |dir_entry| {
+            if (dir_entry.kind != .file) continue;
+            const filename = dir_entry.name;
 
-        // find the snapshots
-        var maybe_latest_full_snapshot: ?FullSnapshotFileInfo = null;
-        var count: usize = 0;
-        for (files) |filename| {
-            const snapshot = FullSnapshotFileInfo.parseFileNameTarZst(filename) catch continue;
-            if (count == 0 or snapshot.slot > maybe_latest_full_snapshot.?.slot) {
-                maybe_latest_full_snapshot = snapshot;
+            if (IncrementalSnapshotFileInfo.parseFileNameTarZst(filename)) |inc_snap| {
+                if (maybe_latest_full) |latest_full| {
+                    if (inc_snap.slot < latest_full.slot) continue;
+                    if (inc_snap.base_slot < latest_full.slot) continue;
+                }
+                try incremental_snaps.append(allocator, inc_snap);
+                continue;
+            } else |_| {}
+
+            const current = FullSnapshotFileInfo.parseFileNameTarZst(filename) catch continue;
+            const prev = maybe_latest_full orelse {
+                maybe_latest_full = current;
+                continue;
+            };
+            if (prev.slot < current.slot) {
+                maybe_latest_full = current;
+                continue;
+            }
+            if (prev.slot == current.slot) {
+                // TODO:
+                std.debug.panic("TODO: report this error gracefully in some way ({s} vs {s})", .{
+                    prev.snapshotArchiveName().constSlice(),
+                    current.snapshotArchiveName().constSlice(),
+                });
             }
-            count += 1;
         }
-        const latest_full_snapshot = maybe_latest_full_snapshot orelse
-            return error.NoFullSnapshotFileInfoFound;
-
-        count = 0;
-        var maybe_latest_incremental_snapshot: ?IncrementalSnapshotFileInfo = null;
-        for (files) |filename| {
-            const snapshot = IncrementalSnapshotFileInfo.parseFileNameTarZst(filename) catch continue;
-            // need to match the base slot
-            if (snapshot.base_slot == latest_full_snapshot.slot and (count == 0 or
-                // this unwrap is safe because count > 0
-                snapshot.slot > maybe_latest_incremental_snapshot.?.slot))
-            {
-                maybe_latest_incremental_snapshot = snapshot;
+        const latest_full = maybe_latest_full orelse return error.NoFullSnapshotFileInfoFound;
+
+        var maybe_latest_incremental: ?IncrementalSnapshotFileInfo = null;
+        for (incremental_snaps.items) |current| {
+            if (current.base_slot != latest_full.slot) continue;
+            const prev = maybe_latest_incremental orelse {
```

Comment:
would prefer to avoid 'prev' and 'current' naming when all we need is the incremental snapshot and the latest incremental snapshot

Reply:
The prev and current naming is useful for quickly understanding the relationship between the two; how about using it as a suffix?

Reply:
We care because one of these variables is the one that is currently selected, the other is the one we're comparing against, naming them too similarly is an easy way to misunderstand code; they can't both be called just "incremental". What would you suggest instead?

Reply:
one would be called `latest_incremental` and the other is called `incremental`

Reply:
"incremental" shadows the method `incremental`; how about `latest_info` and `info`, since we're clearly looping over the "incremental infos"?

Reply:
i feel like this is an okay time to use `_incremental` -- its done elsewhere in the codebase too when naming conflicts happen like this - infos not great because you have to remember what method you are in

Reply:
4ebc727e2269d93e845611ff5083e141176c0dca